x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/Clamp.java

```
@@ -0,0 +1,153 @@
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Clamp", Clamp::new);
+
+    private final Expression field;
+    private final Expression min;
+    private final Expression max;
+
+    @FunctionInfo(
+        returnType = { "double", "integer", "long", "double", "unsigned_long", "keyword", "ip", "boolean", "date", "version" },
+        description = "Clamps the values of all samples to have a lower limit of min and an upper limit of max.",
+        examples = { @Example(file = "k8s-timeseries-clamp", tag = "clamp") }
+    )
+    public Clamp(
+        Source source,
+        @Param(
+            name = "field",
+            type = { "double", "integer", "long", "double", "unsigned_long", "keyword", "ip", "boolean", "date", "version" },
+            description = "Numeric expression. If `null`, the function returns `null`."
+        ) Expression field,
+        @Param(
+            name = "min",
+            type = { "double", "integer", "long", "double", "unsigned_long", "keyword", "ip", "boolean", "date", "version" },
+            description = "The min value to clamp data into."
+        ) Expression min,
+        @Param(
+            name = "max",
+            type = { "double", "integer", "long", "double", "unsigned_long", "keyword", "ip", "boolean", "date", "version" },
+            description = "The max value to clamp data into."
+        ) Expression max
+    ) {
+        super(source, List.of(field, min, max));
+        this.field = field;
+        this.min = min;
+        this.max = max;
+    }
+
+    private Clamp(StreamInput in) throws IOException {
+        this(
+            Source.readFrom((PlanStreamInput) in),
+            in.readNamedWriteable(Expression.class),
+            in.readNamedWriteable(Expression.class),
+            in.readNamedWriteable(Expression.class)
+        );
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
```

I don't think we need to support serialization for this class. Can you throw UnsupportedOperationException and remove the serialization code?